src/main/webapp/scripts/private-room-script.js

```diff
@@ -86,43 +94,72 @@ function onYouTubeIframeAPIReady() {
     });
 }
 
+// Log state changes
+function onStateChange(event) {
+    var state = event.data
+    playerTimeStamp = Math.round(youtubePlayer.getCurrentTime());
+    updateCurrentState(state, playerTimeStamp);
+}
+
+// Every three seconds you listen to youtube player for any detection of change
+window.setInterval(function(){
+    listenForStateChange();
+}, STATE_LISTENER_TIMER_MS);
+
+// This function fetches the state of the private room video and
+// plays/pauses it accordingly
+async function listenForStateChange(){
+    let privateRoomDataPromise = await fetch(SYNC_PATH_WITH_QUERY_PARAM+roomId);
+    // fetch the json-version of the urls for all the youtube videos
+    let privateRoomData = await privateRoomDataPromise.json();
+    // Change timestamp to match group timestamp if client is not within two seconds of room
+    if(Math.abs(playerTimeStamp - privateRoomData.timestamp) >= PLAYER_SECONDS_DISCREPANCY){
+        youtubePlayer.seekTo(privateRoomData.timestamp);
+    }
+    // Change state to match group state
+    if(privateRoomData.currentState === PLAYER_STATE_UNSTARTED) {
+        console.log('Video is Unstarted');
+    } else if(privateRoomData.currentState === PLAYER_STATE_ENDED) {
+        console.log('Video has Ended');
+    } else if(privateRoomData.currentState === PLAYER_STATE_PLAYED){
+        console.log('Video is Playing');
+        playVideo();
+    } else if(privateRoomData.currentState === PLAYER_STATE_PAUSED) {
+        console.log('Video is Paused');
+        pauseVideo();
+    } else if(privateRoomData.currentState === PLAYER_STATE_BUFFERING) {
+        console.log('Video is Buffering');
+    } else if(privateRoomData.currentState === PLAYER_STATE_VIDEO_CUED) {
+        console.log('Video is Cued');
+    } else {
+        console.log('Unknown State');
```

Comment:
Why only logging here?

Reply:
Tianxiao said on a previous comment that she wanted logging even though there were no actions being taken